src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -533,10 +539,10 @@ def enable_zip_deploy(cmd, resource_group_name, name, src, timeout=None, slot=No
                        "config appsettings delete --name MyWebApp --resource-group MyResourceGroup "
                        "--setting-names <setting-names> to delete them.".format(deployment_status_url))
 
-    # check the status of async deployment
-    response = _check_zip_deployment_status(cmd, resource_group_name, name, deployment_status_url,
-                                            authorization, timeout)
-    return response
+    # check if an error occured during deployment
+    if res.status_code:
+        raise CLIError("An error occured during deployment. Status Code: {}, Details: {}"
+                       .format(res.status_code, res.text))
```

Comment:
Could you please use the specific error type from `azure.cli.core.azclierror` instead of `CLIError` ?

Reply:
`AzureInternalError` looked like the best option.